app/code/Magento/InventoryCatalog/Model/GetProductIdsBySkus.php

```diff
@@ -35,15 +40,19 @@ public function __construct(
      */
     public function execute(array $skus): array
     {
-        $idsBySkus = $this->productResource->getProductsIdsBySkus($skus);
-        $notFoundedSkus = array_diff($skus, array_keys($idsBySkus));
+        $cacheKey = hash('md5', implode(',', $skus));
+
+        if (!isset($this->productIdsBySkus[$cacheKey])) {
+            $this->productIdsBySkus[$cacheKey] = $this->productResource->getProductsIdsBySkus($skus);
```

Comment:
This line 
` $this->productIdsBySkus[$cacheKey] = $this->productResource->getProductsIdsBySkus($skus);`
should be after the check for `notFoundedSkus `